### PR TITLE
VTCCA-3978: ensure PII does not appear in URL for agent revoke/appoint

### PR DESCRIPTION
**Pull request by habeeb01**

### CODE REVIEW

File: app/binders/SearchDirection.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  implicit val queryStringBindable: QueryStringBindable[SearchDirection] =
+    new QueryStringBindable[SearchDirection] {
+      override def bind(
+            key: String,
+            params: Map[String, Seq[String]]
+      ): Option[Either[String, SearchDirection]] =
+        params.get(key).collect {
+          case Seq(s) =>
+            SearchDirection.values.find(_.toString == s).toRight("invalid value")
```
Comment (darrell-plant) [inline app/binders/SearchDirection.scala:35]: can this just be `values.find(_.toString == s).toRight("invalid value")` ?

File: app/binders/propertylinks/ExternalPropertyLinkManagementSortField.scala [hunk 2]
```diff
@@ -24,4 +26,19 @@ object ExternalPropertyLinkManagementSortField extends Enumeration {
...
+
+  implicit val queryStringBindable: QueryStringBindable[ExternalPropertyLinkManagementSortField] =
+    new QueryStringBindable[ExternalPropertyLinkManagementSortField] {
+      override def bind(
+            key: String,
+            params: Map[String, Seq[String]]
+      ): Option[Either[String, ExternalPropertyLinkManagementSortField]] =
+        params.get(key).collect {
+          case Seq(s) =>
+            ExternalPropertyLinkManagementSortField.values.find(_.toString == s).toRight("invalid value")
```
Comment (darrell-plant) [inline app/binders/propertylinks/ExternalPropertyLinkManagementSortField.scala:38]: can this just be `values.find(_.toString == s).toRight("invalid value")` ?

File: app/binders/SearchDirection.scala (hunk 1 above)
Comment (PeterPerhac) [inline app/binders/SearchDirection.scala:35]: could the error be a bit more specific? "invalid search direction parameter" or something like that...

File: app/binders/propertylinks/ExternalPropertyLinkManagementSortField.scala (hunk 2 above)
Comment (PeterPerhac) [reply app/binders/propertylinks/ExternalPropertyLinkManagementSortField.scala:38]: could the error be a bit more specific? "invalid sort field parameter" or something like that...

**Review by PeterPerhac — COMMENTED** [review]
👍  looks good

"invalid value" message isn't getting displayed to user anyways, right?

File: app/views/propertyrepresentation/appoint/appointAgentProperties.scala.html [hunk 3]
```diff
@@ -33,7 +34,7 @@
     params: GetPropertyLinksParameters,
     agentCode: Long,
     agentAppointed: Option[String],
-    backLink: Option[String]
+    backLink: Option[String], filterForm: Form[FilterAppointPropertiesForm]
```
Comment (darrell-plant) [inline app/views/propertyrepresentation/appoint/appointAgentProperties.scala.html:37]: `filterForm` is not used

File: app/views/propertyrepresentation/revokeAgentProperties.scala.html [hunk 4]
```diff
@@ -29,7 +29,7 @@
     model: AppointAgentPropertiesVM,
     pagination: PaginationParameters,
     params: GetPropertyLinksParameters,
-    agentCode: Long, backLink: String
+    agentCode: Long, backLink: String, addressForm: Form[String]
```
Comment (darrell-plant) [inline app/views/propertyrepresentation/revokeAgentProperties.scala.html:32]: I don't think `addressForm` is used here

File: build.sbt [hunk 5]
```diff
@@ -85,6 +85,7 @@ val playSettings: Seq[Setting[_]] = Seq(
     "models.SortOrder",
     "models.messages.MessagePagination",
     "models.searchApi.AgentPropertiesParameters",
+    "binders.propertylinks.ExternalPropertyLinkManagementSortField._",
```
Comment (darrell-plant) [inline build.sbt:88]: be nicer about 4 lines up :)

**Comment by darrell-plant** [comment]
LGTDz - probably worth checking why the two forms are not being referenced in the templates :thu: (also, my thumb emoji has stopped working)

File: app/binders/SearchDirection.scala (hunk 1 above)
Comment (habeeb01) [reply app/binders/SearchDirection.scala]: SearchDirection has now been removed as it was not used.

**Comment by darrell-plant** [comment]
LGTDz 👍

**Pull request merged**